### PR TITLE
Support reusable workflows

### PR DESCRIPTION
**Pull request by andreas-aman**
This PR adds support for creating documentation of reusable workflows.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -50,16 +58,24 @@ runs:
           EXTRA_ARGS="${EXTRA_ARGS} --outputColumns ${output_column}"
         done
         
+        for secret_column in ${SECRETS_COLUMNS[@]}; do
+          EXTRA_ARGS="${EXTRA_ARGS} --secretsColumns ${secret_column}"
+        done
+
+        if [[ -n "$REUSABLE" ]]; then
```
Comment (jackton1) [inline action.yml:65]:
```suggestion
        if [[ "$REUSABLE" == “true” ]]; then
```

**Review by jackton1 — CHANGES_REQUESTED** [review]

**Comment by andreas-aman** [comment]
I just realized it must also support outputs from reusable workflows and a bug I introduced.. Will update PR.

**Comment by jackton1** [comment]
@andreas-aman Thanks I’ll take a look later today

**Comment by andreas-aman** [comment]
> @andreas-aman Thanks I’ll take a look later today

@jackton1 Had time to look at this yet?

**Comment by Makeshift** [comment]
I'm currently in need of this feature, but for the life of me I can't figure out how to make @andreas-aman's fork work. There seems to be hardcoded references to tj-actions in the project, and I tried to get go to use my cloned fork by using `go mod edit -replace github.com/tj-actions/auto-doc=/my/cloned/path`, but if I `go build` then it seems to just build the code from this repository as opposed to the forked repository.

Apologies for my newbie-ness when it comes to Go, but I wanted to test this feature out.

**Comment by andreas-aman** [comment]
> 



> I'm currently in need of this feature, but for the life of me I can't figure out how to make @andreas-aman's fork work. There seems to be hardcoded references to tj-actions in the project, and I tried to get go to use my cloned fork by using `go mod edit -replace github.com/tj-actions/auto-doc=/my/cloned/path`, but if I `go build` then it seems to just build the code from this repository as opposed to the forked repository.
> 
> Apologies for my newbie-ness when it comes to Go, but I wanted to test this feature out.

@Makeshift Not sure how to achieve this with the git cli. But if you have the github cli you can simply checkout this PR and build it. `gh pr checkout 405`

File: action.yml [hunk 2]
```diff
@@ -50,16 +62,28 @@ runs:
...
         
+        for output_column in ${REUSABLE_OUTPUT_COLUMNS[@]}; do
+          EXTRA_ARGS="${EXTRA_ARGS} --reusableOutputColumns ${output_column}"
+        done
+
+        for secret_column in ${SECRETS_COLUMNS[@]}; do
+          EXTRA_ARGS="${EXTRA_ARGS} --secretsColumns ${secret_column}"
+        done
+
+        if [[ "$REUSABLE" == “true” ]]; then
```
Comment (Makeshift) [inline action.yml:73]: There's some unicode quotes here, fixed in https://github.com/andreas-aman/auto-doc/pull/2/files. With this change made, this PR seems to work great!

I used the following horrific hack to test it:
```yaml
      - name: Download auto-doc
        id: autodoc
        env:
          GH_TOKEN: ${{ github.token }}
        run: |
          mkdir -p /tmp/auto-doc /tmp/gopath
          oldcd=$(pwd)
          cd /tmp/auto-doc
          git clone https://github.com/tj-actions/auto-doc.git .
          gh pr checkout 405
          go install
          cd "$oldcd"
          rm -rf /tmp/auto-doc
          BIN_PATH="${GOPATH:-$HOME/go}/bin/auto-doc"
          "$BIN_PATH" -h
          echo "BIN_PATH=$BIN_PATH" >> "$GITHUB_OUTPUT"

      - name: Run auto-doc
        uses: Makeshift/auto-doc@reusable # Should be replaced when this PR is merged: https://github.com/tj-actions/auto-doc/pull/405
        with:
          bin_path: ${{ steps.autodoc.outputs.BIN_PATH }}
          filename: '.github/workflows/workflow.yml'
          reusable: true
```

Comment (andreas-aman) [reply action.yml]: I wonder from where these unicode quotes came? Nice catch!

**Review by jackton1 — COMMENTED** [review]
Nice work @andreas-aman @Makeshift. I’m hoping to carve sometime today to review this PR

**Comment by jackton1** [comment]
@all-contributors please add @andreas-aman for code and docs

**Pull request merged**